core/include/cubos/core/al/audio_device.hpp

```diff
@@ -51,36 +43,40 @@ namespace cubos::core::al
         /// @brief Forbid copy construction.
         AudioDevice(const AudioDevice&) = delete;
 
-        /// @brief Creates an audio device from a given device @p specifier.
+        /// @brief Creates an audio device.
         /// @see enumerateDevices()
-        /// @param specifier Device specifier (empty for default).
         /// @return Audio device, or nullptr on failure.
-        static std::shared_ptr<AudioDevice> create(const std::string& specifier = "");
-
-        /// @brief Enumerates the available devices.
-        /// @param[out] devices Vector to fill with the available devices.
-        static void enumerateDevices(std::vector<std::string>& devices);
+        static std::shared_ptr<AudioDevice> create();
 
         /// @brief Creates a new audio buffer
+        /// @param filePath File path to create buffer from.
         /// @return Handle of the new buffer.
-        virtual Buffer createBuffer() = 0;
+        static Buffer createBuffer(const void* data, size_t dataSize);
```

Comment:
Why are these methods static? Don't they depend on a specific implementation?
They should be virtual, so that they are forwarded to the right implementation automatically